internal/imemo.h

```diff
@@ -257,4 +258,60 @@ MEMO_V2_SET(struct MEMO *m, VALUE v)
     RB_OBJ_WRITE(m, &m->v2, v);
 }
 
+struct rb_class_fields {
+    VALUE flags;
+    VALUE klass;
+    union {
+        struct {
+            VALUE fields[1];
+        } embed;
+        struct {
+            VALUE *ptr;
+        } external;
+        struct {
+            st_table *table; // TODO: embed
+        } complex;
+    } as;
+};
+
+#define OBJ_FIELD_EXTERNAL IMEMO_FL_USER0
+#define OBJ_FIELD_COMPLEX  IMEMO_FL_USER1
```

Comment:
It seems like sometimes you check the shape for too complex, and sometimes you check `OBJ_FIELD_COMPLEX` flag. Isn't this duplicate information?